solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3579 @@
+          .withArgs(thirdPartyContract.address)
+      })
+    })
+  })
+
+  describe("unauthorize", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          maintainerProxy
+            .connect(thirdParty)
+            .unauthorize(thirdPartyContract.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner", () => {
+      context("when there are no authorized maintainers", () => {
+        before(async () => {
+          await createSnapshot()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            maintainerProxy
+              .connect(governance)
+              .unauthorize(thirdPartyContract.address)
+          ).to.be.revertedWith("No contract to unauthorize")
+        })
+      })
+
+      context("when there are authorized maintainers", () => {
+        context(
+          "when maintainer to unauthorize is not among the authorized maintainers",
+          () => {
+            before(async () => {
+              await createSnapshot()
+              const signers = await helpers.signers.getUnnamedSigners()
+              const maintainers = [
+                signers[1],
+                signers[2],
+                signers[3],
+                signers[4],
+                signers[5],
+                signers[6],
+                signers[7],
+                signers[8],
+              ]
+
+              for (let i = 0; i < maintainers.length; i++) {
+                /* eslint-disable no-await-in-loop */
+                await maintainerProxy
+                  .connect(governance)
+                  .authorize(maintainers[i].address)
+              }
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should revert", async () => {
+              await expect(
+                maintainerProxy
+                  .connect(governance)
+                  .unauthorize(governance.address)
+              ).to.be.revertedWith("No contract to unauthorize")
+            })
+          }
+        )
+      })
+
+      context("when there is one authorized maintainer", () => {
+        context("when unauthorizing the one that is authorized", () => {
+          let tx: ContractTransaction
+
+          before(async () => {
+            await createSnapshot()
+
+            await maintainerProxy
+              .connect(governance)
+              .authorize(thirdPartyContract.address)
+
+            tx = await maintainerProxy
+              .connect(governance)
+              .unauthorize(thirdPartyContract.address)
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should unauthorize a contract", async () => {
+            await expect(
```

Comment:
The first `await` is not needed.

Reply:
[695eac7](https://github.com/keep-network/tbtc-v2/pull/209/commits/695eac72c524c2877954e301744f09ed2a92bebd)